### PR TITLE
réglage de la diffusion du bouton

### LINKED ISSUE #329: button "retreive budget"/finished adpool
Once the campaign is finished, we have detected display issues related to the "retreive budget" button.

- Precondition:
campaign finished 
The "Retrieve Budget" button becomes clickable only after 15 days from the campaign's end date.

- Issues:
 *When the campaign has ended less than 15 days ago
1)The update of the counter (Retrieve Budget button) will only occur after refreshing the page. 
   The button text should be placed on the same line.
232c2f68-7c43-4478-94bd-6b21023c9fe5
2)Upon opening the campaign, the "Retrieve Budget" button along with the counter appears after a certain period of time.
[retreive.webm](https://github.com/SaTT-Wallet/Frontend/assets/135624031/81b59c27-89d2-4a8b-b83e-f350501911fa)

    *When the campaign has ended over than 15 days ago
    1)The size of the "Retrieve Budget" button does not match the Figma design / the button is mispositioned.
bc12f731-c5f5-4e3a-bfac-0a31fd0b40fc

- Figma design:
Capture88
Capture4768
Please remember to adjust the design of "retrieve Budget" button  for different mobile and desktop devices.

### PR DESCRIPTION
**Pull request by MohamedRayenJmili**
Pull Request Description:

This pull request addresses a few issues related to the display and functionality of the "Retrieve Funds" button. The changes made are as follows:

    Adjusted the button display:
        Modified the CSS styles to ensure that the button is always visible, regardless of scrolling behavior.
        Removed any conditions that were causing the button to be hidden or displayed inconsistently.

    Improved instant display:
        Resolved an issue where the button was not showing immediately upon page load or component initialization.
        Adjusted the code to ensure that the button is displayed instantly without any delay.

    Fixed premature button enablement:
        Addressed a bug where the "Retrieve Funds" button was enabled even before the 15-day time period elapsed.
        Updated the logic to correctly determine when the button should be enabled based on the 15-day timeout.

These changes improve the user experience by ensuring that the button is always displayed, shown instantly, and enabled only when it should be according to the specified time constraints.

Please review and merge this pull request to incorporate the necessary fixes and enhancements.

### CODE REVIEW

File: src/app/campaigns/campaign-details/components/campaign-detail/campaign-detail.component.html [hunk 1]
```diff
@@ -438,13 +438,14 @@ <h1 class="campaign-brand">
...
           </button>
+          <!--*ngIf="campaign.isOwnedByUser && campaign.type === 'finished'  && !loadingData && campaign.budget !== '0'"-->
           <button
           type="button"
           [className]="refundButtonDisable ? 'btn-edit-refunds-pool-disabled' : 'btn-edit-refunds-pool'"
           
           id="btn-edit-refunds-pool"
           (click)="getRefunds(campaign.id)"
-          *ngIf="campaign.isOwnedByUser && campaign.type === 'finished'  && !loadingData && campaign.budget !== '0'"
+          *ngIf="campaign.type === 'finished'  && !loadingData && campaign.budget !== '0'"
```
Comment (hichri-louay) [inline src/app/campaigns/campaign-details/components/campaign-detail/campaign-detail.component.html]: I noticed in your recent changes that the campaign.isOwnedByUser condition has been removed. As per my understanding, this function is meant to be available only for the owner of the campaign.

Could you please help me understand why this condition was removed? Is there another check in place to ensure that the function remains accessible only to the campaign owner?

File: src/app/campaigns/campaign-details/components/campaign-detail/campaign-detail.component.ts [hunk 2]
```diff
@@ -271,27 +271,7 @@ export class CampaignDetailComponent implements OnInit {
       this.getKits();
     }
     
-    setTimeout(() => {
```
Comment (hichri-louay) [inline src/app/campaigns/campaign-details/components/campaign-detail/campaign-detail.component.ts]: I noticed in your recent changes that the setTimeout function has been removed. My understanding is that we were using this timeout as a means to delay the execution until the data from the API was available.

Is there an alternative method in place to handle the asynchronous data fetching from the API?

File: src/app/campaigns/campaign-details/components/campaign-detail/campaign-detail.component.ts [hunk 3]
```diff
@@ -271,27 +271,7 @@ export class CampaignDetailComponent implements OnInit {
...
-
 
-      // CALCULATE THE HOURS :
-      this.dateRefundHours = Math.floor(((this.dateRefund.getTime() - Date.now()) / (1000 * 60 * 60 * 24  ) - this.dateRefundDays ) * 24 )
-
-
-      // CALCULATE MINUTES : 
-      this.dateRefundMinutes = Math.floor(((((this.dateRefund.getTime() - Date.now()) / (1000 * 60 * 60 * 24  ) - this.dateRefundDays ) * 24) - this.dateRefundHours) * 60)
-      this.loadingData = false;
-    }, 3000)
```
Comment (hichri-louay) [inline src/app/campaigns/campaign-details/components/campaign-detail/campaign-detail.component.ts:294]: I noticed in your recent changes that the setTimeout function has been removed. My understanding is that we were using this timeout as a means to delay the execution until the data from the API was available.

Is there an alternative method in place to handle the asynchronous data fetching from the API?

**Comment by MohamedRayenJmili** [comment]
Pull Request Description:

This pull request addresses a few issues related to the display and functionality of the "Retrieve Funds" button. The changes made are as follows:

Adjusted the button display:
    Modified the CSS styles to ensure that the button is always visible, regardless of scrolling behavior.
    Removed any conditions that were causing the button to be hidden or displayed inconsistently.

Improved instant display:
    Resolved an issue where the button was not showing immediately upon page load or component initialization.
    Adjusted the code to ensure that the button is displayed instantly without any delay.

These changes improve the user experience by ensuring that the button is always displayed, shown instantly, and enabled only when it should be according to the specified time constraints.

Please review and merge this pull request to incorporate the necessary fixes and enhancements.

**Review by hichri-louay — APPROVED** [review]

File: src/app/campaigns/campaign-details/components/campaign-detail/campaign-detail.component.ts (hunk 3 above)
Comment (HamdiBenK) [reply src/app/campaigns/campaign-details/components/campaign-detail/campaign-detail.component.ts:294]: yes i am the one who removed it and i have putted the logic in the function that fetch the data, the setTimeout function was causing problems

**Review by HamdiBenK — APPROVED** [review]

**Review by ksibisamir — APPROVED** [review]

**Pull request merged**